test/e2e/vsphere/machines.go

```diff
@@ -0,0 +1,288 @@
+package vsphere
+
+import (
+	"context"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/openshift/api/machine/v1beta1"
+	machinesetclient "github.com/openshift/client-go/machine/clientset/versioned/typed/machine/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	e2e "k8s.io/kubernetes/test/e2e/framework"
+
+	"github.com/openshift/machine-api-operator/pkg/controller/vsphere"
+	util "github.com/openshift/machine-api-operator/test/e2e"
+)
+
+const (
+	machineRole         = "e2e-test"
+	machineReadyTimeout = time.Minute * 6
+)
+
+type MachineTest struct {
+	testCase    string
+	machineName string
+	dataDisks   []v1beta1.VSphereDisk
+	ctx         context.Context
+	cfg         *rest.Config
+	c           *kubernetes.Clientset
+	dc          *dynamic.DynamicClient
+	mc          *machinesetclient.MachineV1beta1Client
+}
+
+type MachineSetTest struct {
+	testCase  string
+	msName    string
+	dataDisks []v1beta1.VSphereDisk
+	ctx       context.Context
+	cfg       *rest.Config
+	c         *kubernetes.Clientset
+	dc        *dynamic.DynamicClient
+	mc        *machinesetclient.MachineV1beta1Client
+}
+
+var _ = Describe("[sig-cluster-lifecycle][OCPFeatureGate:VSphereMultiDisk][platform:vsphere] Managed cluster should", func() {
```

Comment:
You may want to use a Ginkgo Table for the tests in here

```
DescribeTable("<desc>", func(in MachineTest) {
  // Do the test
},
Entry("<desc>", MachineTest{

}),
Entry("<desc>", MachineTest{

}),
...
```

Reply:
I'll try that.  I hadn't tried that before.  I'll see if I can get that implemented shortly.

Reply:
I think I did this correctly.  How does it look now?